### PR TITLE
[addons] Fix data encapsulation violation for C++ PVR Addon API classes.

### PR DESCRIPTION
**Pull request by ksooo**
As discussed here: https://github.com/xbmc/xbmc/pull/18091#issuecomment-648095272 ff.

### CODE REVIEW

**Comment by djp952** [comment]
Thank you @ksooo, the time spent is greatly appreciated.

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/pvr/ChannelGroups.h [hunk 1]
```diff
@@ -36,12 +36,12 @@ namespace addon
 ///@{
 class PVRChannelGroup : public CStructHdl<PVRChannelGroup, PVR_CHANNEL_GROUP>
 {
+friend class CInstancePVRClient;
```
Comment (AlwinEsch) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/pvr/ChannelGroups.h:39]: From code guidlines two spaces before, here and other places.

**Review by AlwinEsch — APPROVED** [review]
Looks good and makes absolutely sense.

Thought also on begin to make so, only then leaved away as the `friend class` normally not liked :smile:.

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/pvr/ChannelGroups.h (hunk 1 above)
Comment (ksooo) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/pvr/ChannelGroups.h:39]: Sure, will fix that. Thanks.

**Comment by ksooo** [comment]
jenkins build with addons please

**Comment by ksooo** [comment]
jenkins build this with addons please

**Comment by ksooo** [comment]
Jenkins addon build errors not caused by this PR.

**Pull request merged**